dsl/src/commonMain/kotlin/it/unibo/collektive/aggregate/api/impl/AggregateContext.kt

```diff
@@ -48,21 +48,27 @@ internal class AggregateContext<ID : Any>(
         initial: Init,
         body: YieldingScope<Field<ID, Init>, Field<ID, Ret>>,
     ): Field<ID, Ret> {
-        val messages = messagesAt<Init>(stack.currentPath())
-        val previous = stateAt(stack.currentPath(), initial)
+        val path = stack.currentPath()
+        val messages = messagesAt<Init>(path)
+        val previous = stateAt(path, initial)
         val subject = newField(previous, messages)
         val context = YieldingContext<Field<ID, Init>, Field<ID, Ret>>()
         return body(context, subject).also {
-            val message = SingleOutboundMessage(it.toSend.localValue, it.toSend.excludeSelf())
-            val path = stack.currentPath()
+            val message = SingleOutboundMessage(
+                it.toSend.localValue,
+                when (it.toSend) {
+                    is SingleOutboundMessage<*, *> -> emptyMap()
```

Comment:
```suggestion
                    is SingletonField<*, *> -> emptyMap()
```

Note: if this suggestion is correct and tests pass there's need for a test.